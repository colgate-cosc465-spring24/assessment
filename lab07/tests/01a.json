{
    "ases" : [
        {
            "number": 1,
            "prefix": "11.0.0.0/8"
        },
        {
            "number": 2,
            "prefix": "12.0.0.0/8"
        }
    ],
    "relationships" : [
        {
            "customer": 1,
            "provider": 2
        }
    ]
}
