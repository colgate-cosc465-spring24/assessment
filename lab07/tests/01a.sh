#!/bin/bash

# Determine path to scripts
SCRIPTPATH=`readlink -f $0`
BASEDIR=`dirname $SCRIPTPATH`
source $BASEDIR/common.sh

SIMULATOR=$BASEDIR/simulator.py
TOPO=`echo $0 | sed 's/sh$/json/'`
EXPECTOUT=`echo $0 | sed 's/sh$/out/'`
ACTUALOUT=/tmp/simulator.out

echo "Your simulator's output:"
$SIMULATOR $TOPO | tee $ACTUALOUT
echo ""
echo "Difference between expected output and your simulator's output:"
diff -U 1000 $EXPECTOUT $ACTUALOUT
