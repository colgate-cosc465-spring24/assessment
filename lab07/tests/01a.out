***** Topology *****
AS 1 (11.0.0.0/8): cust=[]; prov=[2]; peer=[]
AS 2 (12.0.0.0/8): cust=[1]; prov=[]; peer=[]
***** Routes *****
AS 1
	11.0.0.0/8: 
	12.0.0.0/8: 2
AS 2
	11.0.0.0/8: 1
	12.0.0.0/8: 
